File: rust/src/aoc2022/day6.rs
use rdcl_aoc_helpers::input::WithReadLines;
use std::{fs::File, collections::HashSet};

pub fn day6(path: &String) {

    let data:String = File::open(path).read_lines::<String>(1).collect();

    println!("Part 1: {}", tune(&data, 4).unwrap());
    println!("Part 2: {}", tune(&data, 14).unwrap());
}

fn tune(signal: &str, length: usize) -> Option<usize> {
    for i in 0..(signal.len() - length + 1) {
        let d = &signal[i..(i + length)];
        let s: HashSet<char> = d.chars().collect();
        if s.len() == length {
            return Some(i + length);
        }
    }
    None
}

File: python/aoc2022/days/day6.py
def day6():
    with open('../input/input6.txt', 'r') as f:
        data = f.read().split('\n')[0]

    print('Part 1: ', tune(data, 4))
    print('Part 2: ', tune(data, 14))


def tune(signal, length):
    for i in range(len(signal) - (length - 1)):
        d = signal[i:i + length]
        s = set([c for c in d])
        if len(s) == length:
            return i + length
    return None
